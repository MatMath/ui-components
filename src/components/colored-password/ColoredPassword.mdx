---
name: Colored Password
menu: Components
---

import { Playground, Props } from 'docz';
import { ColoredPassword } from './ColoredPassword';

# Colored Password

An easy to read UI Colored Password component. Letters are colorized in black, digits in green and other characters in red.

## Properties

<Props of={ColoredPassword} />

## Usage

### Basic usage

Examples below show usage of the Colored Password

<Playground style={{ preview: { display: 'flex' } }}>
  <ColoredPassword id='labelId' value='D4$hlan3*feat$DuNk-C1ub' />
</Playground>
