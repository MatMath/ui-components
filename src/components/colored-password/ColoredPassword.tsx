import * as React from 'react';
import { getClassNames } from '@utility/cssUtils';

import { getValueSplitInStringCategories } from './helpers';
import styles from './ColoredPassword.module.scss';

export interface ColoredPasswordProps {
  /**
   * The password value to display with colors
   * (letters inherit color, digits in green, special characters in red)
   */
  value: string;
  /**
   * The id to link with when associated to a form label
   */
  id?: string;
}

export const ColoredPassword: React.FC<ColoredPasswordProps> = ({
  id,
  value
}) => {
  const coloredPasswordCategories = getValueSplitInStringCategories(value).map(
    (values, key) => (
      <span
        key={key}
        className={values.category && getClassNames(styles[values.category])}
      >
        {values.text}
      </span>
    )
  );

  return (
    <span id={id} className={getClassNames(styles.password)}>
      {coloredPasswordCategories}
    </span>
  );
};
