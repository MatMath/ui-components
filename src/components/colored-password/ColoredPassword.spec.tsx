import * as React from 'react';
import { mount } from 'enzyme';

import { ColoredPassword } from './ColoredPassword';

describe('<ColoredPassword />', () => {
  it('should render default', () => {
    const coloredPassword = mount(
      <ColoredPassword value='D4$hlan3*feat$DuNk-C1ub' />
    );

    expect(coloredPassword).toMatchSnapshot();
  });

  it('should render correct colors', () => {
    const coloredPassword = mount(<ColoredPassword value='ok42$$' />);

    expect(coloredPassword.find('.letter').text()).toEqual('ok');
    expect(coloredPassword.find('.number').text()).toEqual('42');
    expect(coloredPassword.find('.special').text()).toEqual('$$');
  });
});
