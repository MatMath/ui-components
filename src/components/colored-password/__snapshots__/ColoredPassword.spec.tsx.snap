// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColoredPassword /> should render default 1`] = `
<Component
  value="D4$hlan3*feat$DuNk-C1ub"
>
  <span
    className="password"
  >
    <span
      className="letter"
      key="0"
    >
      D
    </span>
    <span
      className="number"
      key="1"
    >
      4
    </span>
    <span
      className="special"
      key="2"
    >
      $
    </span>
    <span
      className="letter"
      key="3"
    >
      hlan
    </span>
    <span
      className="number"
      key="4"
    >
      3
    </span>
    <span
      className="special"
      key="5"
    >
      *
    </span>
    <span
      className="letter"
      key="6"
    >
      feat
    </span>
    <span
      className="special"
      key="7"
    >
      $
    </span>
    <span
      className="letter"
      key="8"
    >
      DuNk
    </span>
    <span
      className="special"
      key="9"
    >
      -
    </span>
    <span
      className="letter"
      key="10"
    >
      C
    </span>
    <span
      className="number"
      key="11"
    >
      1
    </span>
    <span
      className="letter"
      key="12"
    >
      ub
    </span>
  </span>
</Component>
`;
